### ISSUE TITLE
qwen3.5 sft一段时间后卡住，iteration出现跳变

### ISSUE THREAD

**Issue by Bumble666** [modelscope/ms-swift#8152]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

多机sft训练qwen3.5-122b时，训练到达1/3进度时，会出现卡死，同时发现iteration出现跳变（从35/107直接变为36/36），且可以多次复现：

{'loss': 0.32285908, 'grad_norm': 0.62036353, 'learning_rate': 8.64e-06, 'load_balancing_loss': 1.00936341, 'iteration': '34/107', 'elapsed_time': '2h 12m 6s', 'remaining_time': '4h 43m 37s', 'memory(GiB)': 75.71, 'train_speed(s/it)': 233.114145}

{'loss': 0.3175827, 'grad_norm': 0.59729087, 'learning_rate': 8.52e-06, 'load_balancing_loss': 1.00874186, 'iteration': '35/107', 'elapsed_time': '2h 15m 47s', 'remaining_time': '4h 39m 19s', 'memory(GiB)': 75.71, 'train_speed(s/it)': 232.758482}

{'loss': 0.30566102, 'grad_norm': 0.61037707, 'learning_rate': 8.41e-06, 'load_balancing_loss': 1.00819945, 'iteration': '36/36', 'elapsed_time': '2h 19m 31s', 'remaining_time': '0s', 'memory(GiB)': 75.71, 'train_speed(s/it)': 232.537524}
[INFO:swift] The training of Epoch 1 starts...

Train:  34%|███▎      | 36/107 [2:19:31<4:23:38, 222.80s/it]

已经确认在deepspeed环境变量中设置了MODELSCOPE_CACHE

### How to Reproduce / 如何复现

```
    deepspeed --hostfile=./host.txt \
        --master_addr=$CHIEF_IP \
        --master_port=29500 \
        /xxxx/xxxx/xxxxx/ms-swift/swift/cli/_megatron/sft.py \
        --mcore_model $MODEL_PATH \
        --save_safetensors true \
        --model_type qwen3_5_moe \
        --dataset $DATA_PATH \
        --loss_scale ignore_empty_think \
        --train_type full \
        --val_dataset $VAL_PATH \
        --optimizer_cpu_offload true \
        --use_precision_aware_optimizer true \
        --optimizer_offload_fraction 1.0 \
        --moe_permute_fusion true \
        --moe_grouped_gemm true \
        --moe_shared_expert_overlap true \
        --moe_aux_loss_coeff 1e-3 \
        --micro_batch_size 1 \
        --global_batch_size $BS \
        --moe_expert_capacity_factor 2 \
        --recompute_granularity full \
        --recompute_method uniform \
        --recompute_num_layers 1 \
        --tensor_model_parallel_size 2 \
        --pipeline_model_parallel_size 2 \
        --virtual_pipeline_model_parallel_size 4 \
        --expert_model_parallel_size 8 \
        --expert_tensor_parallel_size 1 \
        --context_parallel_size 1 \
        --freeze_vit true \
        --freeze_aligner true \
        --overlap_grad_reduce true \
        --overlap_param_gather true \
        --num_train_epochs 3 \
        --finetune true \
        --cross_entropy_loss_fusion true \
        --lr $LR \
        --lr_warmup_fraction 0.1 \
        --min_lr $MIN_LR \
        --output_dir $SAVE_PATH \
        --logging_steps 1 \
        --eval_steps 999999 \
        --save_steps $SAVE_INTERVAL \
        --save_strategy steps \
        --max_length $((1024 * 17)) \
        --dataset_num_proc 8 \
        --no_save_optim true \
        --no_save_rng true \
        --sequence_parallel true \
        --packing false \
        --padding_free false \
        --load_from_cache_file true \
        --report_to wandb \
        --wandb_project $WANDB_PROJECT \
        --wandb_exp_name $WANDB_EXP_NAME \
        --attention_backend flash
```

**Comment by Jintao-Huang** [comment]
https://github.com/modelscope/ms-swift/blob/83c707ef15cb727df6ab65f071a3538abb9dc044/swift/megatron/trainers/base.py#L220-L226

调变的由来是这里，但是我奇怪的是，前面的is_finished是如何设置为True的。我猜测是vpp的原因

**Comment by Jintao-Huang** [comment]
复现了，我修复一下
